src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h

```diff
@@ -14,12 +14,34 @@ class cRedstoneWireHandler:
 
 public:
 
-	inline static bool IsDirectlyConnectingMechanism(BLOCKTYPE a_Block)
+	inline static bool IsDirectlyConnectingMechanism(cWorld & a_World, BLOCKTYPE a_Block, Vector3i a_QueryPosition, Vector3i a_Offset)
```

Comment:
This function should just take `a_Block`, `a_BlockMeta` and `a_Offset`, there's no reason to have any side effects depending on a world object.